modules/hal_nxp/usb/usb_host_mcux_drv_port.h

```diff
@@ -0,0 +1,290 @@
+	uint32_t transferSofar;				/*!< Length transferred so far*/
+	host_inner_transfer_callback_t callbackFn;	/*!< Transfer callback function*/
+	void *callbackParam;				/*!< Transfer callback parameter*/
+	usb_host_pipe_t *transferPipe;			/*!< Transfer pipe pointer*/
+	usb_setup_struct_t *setupPacket;		/*!< Set up packet buffer*/
+	uint8_t direction;				/*!< Transfer direction;
+							it's values are USB_OUT or USB_IN*/
+	uint8_t setupStatus;				/*!< Set up the transfer status*/
+	union
+	{
+		uint32_t unitHead;			/*!< xTD head for this transfer*/
+		int32_t transferResult;			/*!< KHCI transfer result */
+	} union1;
+
+	union
+	{
+		uint32_t unitTail;			/*!<xTD tail for this transfer*/
+		uint32_t frame;				/*!< KHCI transfer frame number */
+	} union2;
+
+#if USB_HOST_CONFIG_KHCI
+	uint16_t nakTimeout;				/*!< KHCI transfer NAK timeout */
+	uint16_t retry;					/*!< KHCI transfer retry */
+#endif
+	void *uhc_xfer;
+} usb_host_transfer_t;
+
+/*! @brief Cancel transfer parameter structure */
+typedef struct _usb_host_cancel_param
+{
+	usb_host_pipe_handle pipeHandle;	/*!< Canceling pipe handle*/
+	usb_host_transfer_t *transfer;		/*!< Canceling transfer*/
+} usb_host_cancel_param_t;
+
+/*! @brief USB host controller interface structure */
+typedef struct _usb_host_controller_interface
+{
+	/*! Create a controller instance function prototype*/
+	usb_status_t (*controllerCreate)(
+		uint8_t controllerId,
+		usb_host_handle upperLayerHandle,
+		usb_host_controller_handle *controllerHandle);
+	/*! Destroy a controller instance function prototype*/
+	usb_status_t (*controllerDestory)(
+		usb_host_controller_handle controllerHandle);
+	/*! Open a controller pipe function prototype*/
+	usb_status_t (*controllerOpenPipe)(usb_host_controller_handle controllerHandle,
+				usb_host_pipe_handle *pipeHandle,
+				usb_host_pipe_init_t *pipeInit);
+	/*! Close a controller pipe function prototype*/
+	usb_status_t (*controllerClosePipe)(
+		usb_host_controller_handle controllerHandle,
+		usb_host_pipe_handle pipeHandle);
+	/*! Write data to a pipe function prototype*/
+	usb_status_t (*controllerWritePipe)(usb_host_controller_handle controllerHandle,
+				usb_host_pipe_handle pipeHandle,
+				usb_host_transfer_t *transfer);
+	/*! Read data from a pipe function prototype*/
+	usb_status_t (*controllerReadPipe)(usb_host_controller_handle controllerHandle,
+				usb_host_pipe_handle pipeHandle,
+				usb_host_transfer_t *transfer);
+	/*! Control a controller function prototype*/
+	usb_status_t (*controllerIoctl)(usb_host_controller_handle controllerHandle,
+				uint32_t ioctlEvent,
+				void *ioctlParam);
+} usb_host_controller_interface_t;
+
+/*! @brief USB host instance structure */
+typedef struct _usb_host_instance
+{
+	void *controllerHandle;		/*!< The low level controller handle*/
+	host_callback_t deviceCallback;	/*!< Device attach/detach callback*/
+} usb_host_instance_t;
+
+usb_status_t USB_HostDetachDevice(usb_host_handle hostHandle, uint8_t hubNumber,
```

Comment:
It might be worth adding a comment here highlighting that these functions are called by the USB host driver in the HAL, to make their usage and naming easier to follow